# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/chromium/src/chrome/common/ipc_channel_posix.h
# Commit: 110b2384e7d1
# Full Hash: 110b2384e7d11cd70daa2ea0d06f51926642bdc2
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-06-22 21:29:07
# Regressor Bug: 1706374
# File Overlap Count: 1
# Description:
#   Bug 1706374 - Part 9: Allow reading OtherPid() from an opened IPC::Channel, r=handyman
#   
#   This will be used to handle pre-opened channels more reliably in ports code.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D112774
# ==============================================================================

diff -r ab2b086abbd4 -r 110b2384e7d1 ipc/chromium/src/chrome/common/ipc_channel_posix.h
--- a/ipc/chromium/src/chrome/common/ipc_channel_posix.h	Mon Jun 21 21:53:09 2021 +0000
+++ b/ipc/chromium/src/chrome/common/ipc_channel_posix.h	Mon Jun 21 21:53:09 2021 +0000
@@ -51,6 +51,8 @@
   int GetFileDescriptor() const { return pipe_; }
   void CloseClientFileDescriptor();
 
+  int32_t OtherPid() const { return other_pid_; }
+
   // See the comment in ipc_channel.h for info on Unsound_IsClosed() and
   // Unsound_NumQueuedMessages().
   bool Unsound_IsClosed() const;